### PR TITLE
Diff current values against new values to determine if a change event is warranted.

### LINKED ISSUE #105: should .set("values",[]) trigger a "values" event if the value is really the same
If you call:

`attr.set(el, "values",[])`

twice, a "values" event is still fired .. indicating a change in the property.  Is this correct behavior?

### PR DESCRIPTION
**Pull request by Alfredo-Delgado**
resolves #105

### CODE REVIEW

File: dom/attr/attr.js [hunk 1]
```diff
@@ -356,7 +364,23 @@ var formElements = {"INPUT": true, "TEXTAREA": true, "SELECT": true},
 					setupMO(this, function(){
 						var lastVal = setData.get.call(this, "valuesLastVal");
```
Comment (Alfredo-Delgado) [inline dom/attr/attr.js]: When this evaluates, the values returned aren't really the "lastVal" so much as the new values that have been applied. It's possible I'm missing something, but I'd like to change the name of this variable, too.

File: dom/attr/attr.js [hunk 2]
```diff
@@ -344,6 +344,14 @@ var formElements = {"INPUT": true, "TEXTAREA": true, "SELECT": true},
 				},
 				set: function(values){
 					values = values || [];
+
+          // capture current selectedOptions
```
Comment (matthewp) [inline dom/attr/attr.js]: Looks like a tabs v spaces (fight!) issue here. Might want to install an editorconfig plugin in your editor.

File: dom/attr/attr.js (hunk 1 above)
Comment (matthewp) [reply dom/attr/attr.js]: Yeah, let's rename that to `valuesCurVal`, I think that's what it really means.

**Comment by matthewp** [comment]
I don't think this is quite the right approach. 

What we *really* want to do is to compare the values since the last time a dispatch happen. So what I would do is in the two places where we dispatch, save the current state like so:

```js
var curVal = setData.get.call(this, "valuesCurVal");
var lastDispatchedVal = setData.get.call(this, "valuesLastDispatchedVal");

if(!areTheSame(curVal, lastDispatchedVal)) {
  setData.set.call(this, curVal.splice(), / * splice so we get a copy and not a reference */
    "valuesLastDispatchedVal");
}
```

Something like that.

**Comment by justinbmeyer** [comment]
btw, can-util/js/diff can probably help tell if there are differences.

**Pull request merged**